trieste/acquisition/function.py

```diff
@@ -1365,3 +1365,29 @@ def repulsion_term(
             )
 
     return acquisition
+
+
+class PredictiveVariance(SingleModelAcquisitionBuilder):
+    """
+    Builder for the determinant of the predictive covariance matrix over the batch points.
+    For a batch of size 1 it is the same as maximizing the predictive variance.
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return "PredictiveVariance()"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: Unused.
+        :param model: The model over the specified ``dataset``.
+        :return: The determinant of the predictive function.
+        """
+        return lambda at: self.acquisition(model, at)
+
+    @staticmethod
+    def acquisition(model: ProbabilisticModel, at: TensorType) -> TensorType:
+        _, covariance = model.predict_joint(at)
+        return tf.linalg.det(covariance)
```

Comment:
any specific need for this public method? this could be a function in `prepare_acquisition_function`

Reply:
no actually, yeah that'll be neat to make it a function inside prepare_acqusition_function

Reply:
correct me if I am wrong, but making this public method is needed for test code no?
for example in [negative lcb](https://github.com/secondmind-labs/trieste/blob/4ef8d1317defca77b8f0dceed9ccefee894daf8f/tests/unit/acquisition/test_function.py#L375) test.

Reply:
when you call `prepare...` you essentially get `determinantcovariance` back immediately, so you can test via `prepare...`, with  a stub model with custom `predict_joint`
```
class Model(ProbabilisticModel):
    def predict_joint(x):
        return <some function of x that helps you know this isn't being used>, <some function of x that makes it easy to test>
```
and use any old data (you could use some data that gives confidence it's not being used, but I'm not sure that's necessary).

i'm a big fan of keeping as much out of the public API as possible. It helps keep the intention of the API clear. In some cases, we exposed the underlying acquisition function as well as the `AcquisitionFunctionBuilder`. That might be because users might not want to do the work the builder does in the way the builder does it (EI can use different "best" points, for example), or it could be for ease of use (people are perhaps more used to thinking about LCB than negative LCB). In some cases, it may have just been a mistake. I hope we never did it to help with testing. I don't think it's good practice to make things public solely for testing purposes

Reply:
Ah yeah, that makes sense. yeah, I think that'll be the best approach here. Thanks for the suggestion!